pkg/controller/disruption/disruption.go

```diff
@@ -916,7 +916,7 @@ func countHealthyPods(pods []*v1.Pod, disruptedPods map[string]metav1.Time, curr
 		if disruptionTime, found := disruptedPods[pod.Name]; found && disruptionTime.Time.Add(DeletionTimeout).After(currentTime) {
 			continue
 		}
-		if apipod.IsPodReady(pod) {
+		if apipod.IsPodReady(pod) && apipod.ArePodReadinessGatesTrue(pod) {
```

Comment:
Based on the description of [Pod Ready++](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/580-pod-readiness-gates#implementation-detailsnotesconstraints) this check only makes sure this is true, and should be a safe change.

Reply:
Just so I understand the intent of this - the pod isn't ready via the API (via Pod Status having the condition PodReady) unless all the gates are also ready.  This code at the control plane is looking at the same signals and checking that again?  Why do we not trust the kubelet to perform those checks?

I.e. in `pkg/kubelet/status/generate.go`, 

```
// GeneratePodReadyCondition returns "Ready" condition of a pod.
// The status of "Ready" condition is "True", if all containers in a pod are ready
// AND all matching conditions specified in the ReadinessGates have status equal to "True".
func GeneratePodReadyCondition(spec *v1.PodSpec, conditions []v1.PodCondition, containerStatuses []v1.ContainerStatus, podPhase v1.PodPhase) v1.PodCondition {
```

ensures PodReady cannot be true if a readiness gate is false.

Reply:
That was my understanding, and thus considered this a safe change. I'd be curious to hear from @K8Y about the actual use cases she's trying to solve with this.

Reply:
Our discovery of the behavior was during a stuck rolling update which involved a surge of pods. We didn't realize that the PDB was operating off of the deployment’s desired replica count rather than the number of pods across all of the deployment’s replicasets. Sorry for the inconvenience!